test/end2end/app_test.dart

```diff
@@ -84,6 +90,30 @@ void main() {
     await createPost(tester);
     await deletePost(tester);
   });
+
+  testWidgets("Logout from challenges does not cause error", (tester) async {
```

Comment:
I'm not sure this test really has its place in this file. This is not an end-to-end test, or part of one.

Reply:
I refactored the e2e test folder, you can have a look.

Reply:
I'm not sure I get your refactor. This is not an end-to-end test, is this? Moreover, is this refactor really meaningful in this PR? Shouldn't this be another PR?

Reply:
Ok I changed the test to be just on the challenge view-model. I left some positive modifications to the end to end test that I made on the way, but it is just a small refactor that is necessary anyway if we want to add other tests. I think we can keep this in this pr.

Reply:
Yeah, I'm not sure it has entierely its place in this PR, but I guess this is fine since this is already done and probably had to be done at some point. This is much better, thank you :)